HIP/hip-1068.md

```diff
@@ -0,0 +1,528 @@
+2. The enterprise DApp operator configures the Crypto intermediary as their Fee Sponsor for all HBAR transaction 
+    or query costs.
+    1. E.g. Client.setFeeSponsor()
+3. The enterprise DApp operator submits a signed transaction to the network, the transaction body is interrogated
+    (by the network) for the fee_sponsor_account_id and the Fee Sponsors account identified (where present).
+4. The Fee Sponsor’s Account is interrogated (by the network) for a Fee Allowance using the enterprise DApp operators
+    account Id
+    1. The transaction succeeds if the allowance is present and has either:
+        1. unlimited remaining allowance; or
+        2. A remaining balance greater than the current transaction fee
+    2. Otherwise the transaction fails.
+5. Where the transaction succeeds, the HBAR fees for the transaction are deducted from the Fee Sponsors account.
+    1. The enterprise DApp operators HBAR account balance (used to sign the transaction) is left untouched
+    2. The enterprise DApp operator’s HBAR balance can be zero.
+6. The transaction is successful.
+
+### Enterprise Software Provider
+
+User Story: As a software provider I want to cover the transaction fees of a customer’s account to allow them to
+interact with my Software even with 0 HBARs in their account.
+
+User Journey:
+
+1. The software provider signs an AccountAllowanceApproveTransaction (using a new method of approveHbarFeeAllowance)
+    which designates the customer as the spender.
+    1. The fee for this transaction is covered by the software provider.
+    2. The new approveHbarFeeAllowance method could allow for both fixed and unlimited values.
+2. The software provider sets the customer’s fee sponsor as the software provider’s dedicated fee account (and
+    synchronizes this with customer’s software for use during transaction submission).
+3. The customer submits a signed transaction to the network, the transaction body is interrogated (by the network) for
+    the fee_sponsor_account_id and the Fee Sponsors account identified (where present).
+4. The Fee Sponsor’s Account is interrogated (by the network) for a Fee Allowance using the customer’s account Id.
+    1. The transaction succeeds if the allowance is present and has either:
+        1. unlimited remaining allowance; or
+        2. A remaining balance greater than the current transaction fee
+    2. Otherwise the transaction fails.
+5. Where the transaction succeeds, the HBAR fees for the transaction are deducted from the Fee Sponsors account (i.e.
+    the software provider).
+    1. The customer’s HBAR account balance (used to sign the transaction) is left untouched
+    2. The customer’s HBAR balance can be zero.
+6. The transaction is successful.
+
+### Protobuf
+
+A new new allowance scope will be added to help define which type of fee an allowance can be used for
+```protobuf
+    enum AllowanceScope {
```

Comment:
The `GAS` scope seems to be missing.

Reply:
Isn't GAS part of the TRANSACTION_FEE for a Smart contract call?

Reply:
There is an interesting case that we should make sure is covered properly by this scope concept.

If we performa a precompile smart contract call that creates a token - the precompile requires gas amount in tinybars + separate value sent in tinybars, since the gas amount will be above some of the limits. This implies that calling createFungibleToken precompile call might make use of both `GAS` and `TRANSACTION_FEE` scopes.

We can elaborate more of what `TRANSACTION_FEE` covers in this HIP as a scope? I understand it as hbar amount needed to cover transaction expenses different than the gas itself. Or is it actually the gas amount itself?

Reply:
But at some other points GAS is used as a scope (Mirror Node endpoint, for example). Here it must be clear what is the difference and if GAS must be included.

Reply:
I agree, I'm a bit confused regarding the difference between `GAS` and `TRANSACTION_FEE` scope.

Reply:
To clarify, gas is a concept of resource utilization. 
Gas fee is how much each unit of gas costs a user in HBAR.
Here the allowance is capturing the cost in hbar when settlement occurs